prow/config/secret/agent.go

```diff
@@ -108,22 +112,31 @@ func (a *Agent) reloadSecret(secretPath string) {
 }
 
 // GetSecret returns the value of a secret stored in a map.
-func (a *Agent) GetSecret(secretPath string) []byte {
+func (a *Agent) GetSecret(secretPath string) ([]byte, uint32) {
```

Comment:
Rather than changing this method's signature, how about using a subscriber pattern?